client/app/authenticators/yggdrasil.js

```diff
@@ -0,0 +1,66 @@
+import Ember from 'ember';
+import ENV from 'vilive/config/environment';
+import Base from 'ember-simple-auth/authenticators/base';
+
+const { RSVP, run } = Ember;
+
+export default Base.extend({
+  /*
+    https://github.com/simplabs/ember-simple-auth/blob/1.0.1/addon/authenticators/base.js#L86
+    For now just assuming it's valid so the sesion isn't invalidated.
+
+    Maybe in the future we have some route that verifies tokens,
+    but I'm not sure it matters becuase if the hits a api that requires
+    a token and it's invalid it will issue a 401 which will be handled
+  */
+  restore(data) {
+    return new RSVP.Promise((resolve) => {
+      resolve(data);
+    });
+  },
+
+  /**
+   * changed it up a bit, now it just takes a in request
+   * that is constructed in the session manager.
+   *
+   * don't care how it was constructed but this request will return
+   * the authentication token and username.
+   */
+  authenticate(request) {
```

Comment:
This is much simpler now, takes in ajax promise and handles the response and no longer tasked with creating the request. It assumed the request given will have a response with a username/token and will store that in session.